File: frontend/src/api/movement.ts
import { apiClient } from './client';

export interface Movement {
  id: string;
  name: string;
  name_en?: string;
  category?: string;
  muscle_group?: string;
  secondary_muscles?: string[];
  equipment?: string;
  difficulty?: string;
  description?: string;
  instructions?: string[];
  key_points?: string[];
  common_mistakes?: string[];
  video_url?: string;
  contraindications?: string[];
}

export interface MovementFilters {
  muscle_group?: string;
  category?: string;
  difficulty?: string;
  equipment?: string;
  search?: string;
}

export interface MovementCategories {
  muscle_groups: string[];
  categories: string[];
  equipment: string[];
}

export function getMovements(filters?: MovementFilters) {
  return apiClient.get<Movement[]>('/movements', {
    params: filters
  });
}

export function getMovementCategories() {
  return apiClient.get<MovementCategories>('/movements/categories');
}

export function getMovement(id: string) {
  return apiClient.get<Movement>(`/movements/${id}`);
}

File: backend/schemas/movement.py
"""
movementlibSchema
"""

from typing import Optional, List
from pydantic import BaseModel


class MovementResponse(BaseModel):
    id: str
    name: str
    name_en: Optional[str]
    category: Optional[str]
    muscle_group: Optional[str]
    secondary_muscles: Optional[List[str]]
    equipment: Optional[str]
    difficulty: Optional[str]
    description: Optional[str]
    instructions: Optional[List[str]]
    key_points: Optional[List[str]]
    common_mistakes: Optional[List[str]]
    video_url: Optional[str]
    contraindications: Optional[List[str]]

    class Config:
        from_attributes = True


class MovementCreate(BaseModel):
    name: str
    name_en: Optional[str] = None
    category: str
    muscle_group: str
    secondary_muscles: Optional[List[str]] = None
    equipment: str
    difficulty: str
    description: Optional[str] = None
    instructions: Optional[List[str]] = None
    key_points: Optional[List[str]] = None
    common_mistakes: Optional[List[str]] = None
    video_url: Optional[str] = None
    contraindications: Optional[List[str]] = None
